server-mirror-git/src/main/java/com/linecorp/centraldogma/server/internal/mirror/SshGitMirror.java

```diff
@@ -170,27 +145,37 @@ private SshClient createSshClient() {
         builder.serverKeyVerifier((clientSession, remoteAddress, serverKey) -> true);
         builder.randomFactory(() -> bounceCastleRandom);
         final SshClient client = builder.build();
-        configureCredential(client);
-        client.start();
-        return client;
+        try {
+            configureCredential(client);
+            client.start();
+            return client;
+        } catch (Throwable t) {
+            client.stop();
+            throw t;
+        }
     }
 
-    private static ClientSession createSession(SshClient sshClient, URIish uri) {
+    @VisibleForTesting
+    static ClientSession createSession(SshClient sshClient, URIish uri) {
+        int port = uri.getPort();
+        if (port <= 0) {
+            port = 22; // Use the SSH default port it unspecified.
+        }
+        logger.trace("Connecting to {}:{}", uri.getHost(), port);
+        ClientSession session = null;
         try {
-            int port = uri.getPort();
-            if (port <= 0) {
-                port = 22; // Use the SSH default port it unspecified.
-            }
-            logger.trace("Connecting to {}:{}", uri.getHost(), port);
-            final ClientSession session = sshClient.connect(uri.getUser(), uri.getHost(), port)
-                                                   .verify(GitModuleProperties.CONNECT_TIMEOUT.getRequired(
-                                                           sshClient))
-                                                   .getSession();
+            session = sshClient.connect(uri.getUser(), uri.getHost(), port)
+                               .verify(GitModuleProperties.CONNECT_TIMEOUT.getRequired(
+                                       sshClient))
+                               .getSession();
             session.auth().verify(GitModuleProperties.AUTH_TIMEOUT.getRequired(session));
             logger.trace("The session established: {}", session);
             return session;
-        } catch (IOException e) {
-            throw new RuntimeException(e);
+        } catch (Throwable t) {
+            if (session != null) {
+                session.close(true);
```

Comment:
Understood that this is the actual point that is expected to fix the start/stop logs

Reply:
That is correct. 👍